### PR TITLE
Add Kafka Trigger template

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
Create this PR for start conversation with code.  I added Kafka trigger template with https://github.com/Azure/azure-functions-extension-bundles/pull/49 

Since this is the first time contribution, so that I have several questions. 

# 1. Action Id 
Where and how can I get the Action Id? Generate uniq id by myself? 
https://github.com/Azure/azure-functions-templates/blob/tsushi/kafka/Functions.Templates/Templates/KafkaTrigger-CSharp/.template.config/template.json#L45

# 2. Package 
Where can I get the Package ? 
https://github.com/Azure/azure-functions-templates/blob/dev/Functions.Templates/Templates/EventHubTrigger-CSharp/.template.config/vs-2017.3.host.json#L4

# 3. Done
Test v1 and v2 template.  For the test cases 
[TestCases](https://microsoft-my.sharepoint.com/:x:/p/tsushi/EZhAiNbB7vxDmvAE1QctmRsBAufdJCJdBbRg-rEvsEFpzA?e=V6LxRb)

# 3. Next step

* Merge this PR
*  Release this code for testing
* Create PR for extension bundle

### CODE REVIEW

**Comment by TsuyoshiUshio** [comment]
Hi @soninaren 
I tested with func command for non-csharp implemantation.  
For dotnet, I tested with vs and  func with dotnet new command.  [TestCase](https://microsoft-my.sharepoint.com/:x:/p/tsushi/EZhAiNbB7vxDmvAE1QctmRsBAufdJCJdBbRg-rEvsEFpzA?e=EpTcLY)
Now I'm satisfied the outcome.  Only question is ActivityId and Resource Id that I explained on the PR description.

File: Functions.Templates/Templates/KafkaTrigger-CSharp/KafkaTriggerCSharp.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
```
Comment (soninaren) [inline Functions.Templates/Templates/KafkaTrigger-CSharp/KafkaTriggerCSharp.cs:2]: Can this file be split into 2 individual templates?

File: Functions.Templates/Templates/KafkaTrigger-CSharp/metadata.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+    "defaultFunctionName": "KafkaTrigger",
+    "description": "$KafkaTrigger_description",
+    "name": "Kafka trigger",
+    "language": "C#",
+    "category": [
+        "$temp_category_core",
+        "$temp_category_dataProcessing"
+    ],
+    "categoryStyle": "Kafka",
```
Comment (soninaren) [inline Functions.Templates/Templates/KafkaTrigger-CSharp/metadata.json:10]: There is  an invalid value `"categoryStyle": "Kafka",`. No such category.

File: Functions.Templates/Templates/KafkaTrigger-TypeScript/metadata.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "defaultFunctionName": "KafkaTrigger",
+  "description": "$KafkaTrigger_description",
+  "name": "Kafka trigger",
+  "language": "TypeScript",
+  "category": [
+      "$temp_category_core",
+      "$temp_category_dataProcessing"
+  ],
+  "categoryStyle": "Kafka",
+  "enabledInTryMode": true,
```
Comment (soninaren) [inline Functions.Templates/Templates/KafkaTrigger-TypeScript/metadata.json:11]: This should be disabled

**Review by soninaren — CHANGES_REQUESTED** [review]

File: Build/PackageFiles/ExtensionBundleTemplates-1.x.nuspec [hunk 4]
```diff
@@ -34,6 +34,12 @@
     <file src="../../Functions.Templates/templates/RabbitMQTrigger-CSharp/RabbitMQTriggerCSharp.csx" target="templates/RabbitMQTrigger-CSharp/run.csx" />
     <file src="../../Functions.Templates/templates/RabbitMQTrigger-CSharp/readme.md" target="templates/RabbitMQTrigger-CSharp/readme.md" />
     <file src="../../Functions.Templates/templates/RabbitMQTrigger-CSharp/sample.dat" target="templates/RabbitMQTrigger-CSharp/sample.dat" />
+    <file src="../../Functions.Templates/templates/KafkaTrigger-CSharp/sample.dat" target="templates/KafkaTrigger-CSharp/sample.dat" />
```
[the remark this answers is not in the document]
Comment (TsuyoshiUshio) [reply Build/PackageFiles/ExtensionBundleTemplates-1.x.nuspec]: Done

File: Build/PackageFiles/ExtensionBundleTemplates-1.x.nuspec [hunk 5]
```diff
@@ -62,10 +68,18 @@
     <file src="../../Functions.Templates/templates/RabbitMQTrigger-JavaScript/metadata.json" target="templates/RabbitMQTrigger-JavaScript/metadata.json" />
     <file src="../../Functions.Templates/templates/RabbitMQTrigger-JavaScript/readme.md" target="templates/RabbitMQTrigger-JavaScript/readme.md" />
     <file src="../../Functions.Templates/templates/RabbitMQTrigger-JavaScript/sample.dat" target="templates/RabbitMQTrigger-JavaScript/sample.dat" />
+    <file src="../../Functions.Templates/templates/KafkaTrigger-JavaScript/function.json" target="templates/KafkaTrigger-JavaScript/function.json" />
+    <file src="../../Functions.Templates/templates/KafkaTrigger-JavaScript/index.js" target="templates/KafkaTrigger-JavaScript/index.js" />
+    <file src="../../Functions.Templates/templates/KafkaTrigger-JavaScript/metadata.json" target="templates/KafkaTrigger-JavaScript/metadata.json" />
+    <file src="../../Functions.Templates/templates/KafkaTrigger-JavaScript/readme.md" target="templates/KafkaTrigger-JavaScript/readme.md" />
+    <file src="../../Functions.Templates/templates/KafkaTrigger-JavaScript/sample.dat" target="templates/KafkaTrigger-JavaScript/sample.dat" />
+
```
[the remark this answers is not in the document]
Comment (TsuyoshiUshio) [reply Build/PackageFiles/ExtensionBundleTemplates-1.x.nuspec]: Done

File: Functions.Templates/Templates/KafkaTrigger-CSharp/metadata.json (hunk 2 above)
Comment (TsuyoshiUshio) [reply Functions.Templates/Templates/KafkaTrigger-CSharp/metadata.json]: Done

Comment (TsuyoshiUshio) [reply Functions.Templates/Templates/KafkaTrigger-CSharp/metadata.json]: I made it as "other" It it what you expected?

File: Functions.Templates/Templates/KafkaTrigger-TypeScript/metadata.json (hunk 3 above)
Comment (TsuyoshiUshio) [reply Functions.Templates/Templates/KafkaTrigger-TypeScript/metadata.json]: Done

File: Functions.Templates/Templates/KafkaTrigger-CSharp/KafkaTriggerCSharp.cs (hunk 1 above)
Comment (TsuyoshiUshio) [reply Functions.Templates/Templates/KafkaTrigger-CSharp/KafkaTriggerCSharp.cs]: Sure. I removed the KafkaOutput I'll add the KafkaOutput with following pull requests in small sized ones.

File: Functions.Templates/Templates/KafkaTrigger-CSharp/metadata.json (hunk 2 above)
Comment (TsuyoshiUshio) [reply Functions.Templates/Templates/KafkaTrigger-CSharp/metadata.json]: Yes.

File: Functions.Templates/Templates/KafkaTrigger-CSharp/KafkaTriggerCSharp.cs (hunk 1 above)
Comment (TsuyoshiUshio) [reply Functions.Templates/Templates/KafkaTrigger-CSharp/KafkaTriggerCSharp.cs]: Done. I'll separate the KafkaOutput. I'll send via separate small prs.

**Comment by TsuyoshiUshio** [comment]
Hi @pragnagopa  This is the PR.

**Comment by pragnagopa** [comment]
@soninaren  - gentle ping! Is the PR ready to merge?

**Review by soninaren — APPROVED** [review]

**Pull request merged**